### PR TITLE
Add listTable() tests

### PR DESCRIPTION
**Pull request by MGatner**
**Description**
Currently there are no tests for the database metadata methods. There are a lot more that should be added but this creates the file and adds tests for `listTables()` that exhibits the error described in https://github.com/codeigniter4/CodeIgniter4/issues/2210.

Someone who knows what's up with the weird `!` escape character business take a look?

**Checklist:**
- [X] Securely signed commits
- [X] Component(s) with PHPdocs
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [X] Conforms to style guide

### CODE REVIEW

**Comment by MGatner** [comment]
@lonnieezell Sorry I made a mess of this. Good news is `[ci skip]` worked on my commit above!

I believe this is correct now and the failure demonstrates the bug with `ConstrainPrefix`. I'm still not sure on a solution.

File: tests/system/Database/Live/MetadataTest.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+		// Restore the prefix and get the tables with the original prefix
+		$this->db->setPrefix($DBPrefix);
+		$result = $this->db->listTables(true);
+
+		$this->assertEquals($this->expectedTables, array_values($result));
+
+		// Clean up temporary table
+		$this->db->setPrefix('tmp_');
+		$forge->dropTable('widgets');
```
Comment (michalsn) [inline tests/system/Database/Live/MetadataTest.php]:
```suggestion
		$this->forge->dropTable('widgets');
```

Comment (MGatner) [reply tests/system/Database/Live/MetadataTest.php]: Good catch, thanks for that!

**Comment by MGatner** [comment]
How the 🤬 did code coverage go *down* then all I did was add tests?? @lonnieezell this is ready to go unless @michalsn notices anything else.

**Pull request merged**